File: bindings/python/examples/get_object.py
# Copyright (c) 2025 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from lib.iota_sdk import *

import asyncio


async def main():
    client = GraphQlClient.new_testnet()

    object_id = ObjectId.from_hex(
        "0x541b117cac18fb1c07a293db300acd12b05c01fa81232b37151b005ca7d4f755")

    obj = await client.object(object_id)
    if obj is None:
        raise Exception("missing object")

    print("Object ID:", obj.id().to_hex())
    print("Version:", obj.version())
    print("Previous transaction:", obj.previous_transaction().to_base58())
    print("Owner:", obj.owner())
    print("Storage rebate:", obj.storage_rebate())
    print("Type:", obj.object_type())
    print("BCS bytes:", hex_encode(obj.as_struct().contents))


if __name__ == "__main__":
    asyncio.run(main())

File: bindings/swift/examples/GetObject.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import Foundation
import IotaSDK

@main
struct GetObjectExample {
  static func main() async throws {
    let client = GraphQlClient.newTestnet()

    let objectId = try ObjectId.fromHex(
      hex: "0x541b117cac18fb1c07a293db300acd12b05c01fa81232b37151b005ca7d4f755")

    guard let obj = try await client.object(objectId: objectId) else {
      throw NSError(
        domain: "GetObject", code: 1,
        userInfo: [NSLocalizedDescriptionKey: "missing object"])
    }

    print("Object ID:", obj.id().toHex())
    print("Version:", obj.version())
    print("Previous transaction:", obj.previousTransaction().toBase58())
    print("Owner:", obj.owner())
    print("Storage rebate:", obj.storageRebate())
    print("Type:", obj.objectType())
    print("BCS bytes:", hexEncode(input: obj.asStruct().contents))
  }
}
